src/main/java/org/csanchez/jenkins/plugins/kubernetes/pipeline/PodTemplateStepExecution.java

```diff
@@ -45,24 +50,28 @@
 
     @Override
     public boolean start() throws Exception {
-
         Cloud cloud = Jenkins.getInstance().getCloud(cloudName);
         if (cloud == null) {
             throw new AbortException(String.format("Cloud does not exist: %s", cloudName));
         }
         if (!(cloud instanceof KubernetesCloud)) {
-            throw new AbortException(String.format("Cloud is not a Kubernetes cloud: %s (%s)", cloudName,
-                    cloud.getClass().getName()));
+            throw new AbortException(
+                    String.format("Cloud is not a Kubernetes cloud: %s (%s)", cloudName, cloud.getClass().getName()));
         }
         KubernetesCloud kubernetesCloud = (KubernetesCloud) cloud;
 
         Run<?, ?> run = getContext().get(Run.class);
+        if (kubernetesCloud.isUsageRestricted()) {
+            checkAccess(run, kubernetesCloud);
+        }
+
         PodTemplateAction podTemplateAction = run.getAction(PodTemplateAction.class);
         NamespaceAction namespaceAction = run.getAction(NamespaceAction.class);
         String parentTemplates = podTemplateAction != null ? podTemplateAction.getParentTemplates() : null;
 
-        //Let's generate a random name based on the user specified to make sure that we don't have
-        //issues with concurrent builds, or messing with pre-existing configuration
+        // Let's generate a random name based on the user specified to make sure that we
```

Comment:
Remove formatting change (older is better)

Reply:
fixed